src/main/java/com/semmle/jira/addon/util/JiraUtils.java

```diff
@@ -138,4 +148,63 @@ public static void createLgtmIssueType() {
       // Issue Type already created
     }
   }
+
+  public static void createConfigKeyCustomField() throws GenericEntityException {
+
+    CustomFieldManager customFieldManager = ComponentAccessor.getCustomFieldManager();
+
+    ManagedConfigurationItemService managedConfigurationItemService =
+        ComponentAccessor.getComponentOfType(ManagedConfigurationItemService.class);
+
+    CustomField customField = null;
+    ManagedConfigurationItem managedField;
+
+    for (CustomField candidateField :
+        customFieldManager.getCustomFieldObjectsByName(Constants.CUSTOM_FIELD_NAME)) {
+
+      managedField = managedConfigurationItemService.getManagedCustomField(candidateField);
+
+      if (managedField != null
+          && managedField.isManaged()
+          && managedField.getConfigurationItemAccessLevel() == ConfigurationItemAccessLevel.LOCKED
+          && managedField.getSourceId() == Constants.PLUGIN_KEY) {
+        customField = candidateField;
+        break;
+      }
+    }
+
+    if (customField == null) {
+
+      customField =
+          customFieldManager.createCustomField(
+              Constants.CUSTOM_FIELD_NAME,
+              Constants.CUSTOM_FIELD_NAME,
+              customFieldManager.getCustomFieldType(
+                  "com.atlassian.jira.plugin.system.customfieldtypes:textfield"),
+              customFieldManager.getCustomFieldSearcher(
+                  "com.atlassian.jira.plugin.system.customfieldtypes:exacttextsearcher"),
+              Collections.singletonList(GlobalIssueContext.getInstance()),
+              Collections.singletonList(JiraUtils.getLgtmIssueType()));
+
+      managedField = managedConfigurationItemService.getManagedCustomField(customField);
+
+      ManagedConfigurationItemBuilder builder =
+          ManagedConfigurationItemBuilder.builder(managedField);
+
+      builder.setManaged(true);
+      builder.setConfigurationItemAccessLevel(ConfigurationItemAccessLevel.LOCKED);
+      builder.setSource(Constants.PLUGIN_KEY);
+
+      managedConfigurationItemService.updateManagedConfigurationItem(builder.build());
+    }
+
+    PluginSettingsFactory pluginSettingsFactory =
+        ComponentAccessor.getOSGiComponentInstanceOfType(PluginSettingsFactory.class);
```

Comment:
Did the normal `getComponent` not work?

Reply:
No, didn't seem to. Not sure why...

Reply:
This looks like a good explanation: https://community.atlassian.com/t5/Answers-Developer-Questions/PluginSettingsFactory-missing/qaq-p/502891